awx/main/analytics/core.py

```diff
@@ -52,7 +52,7 @@ def all_collectors():
     }
 
 
-def register(key, version, description=None, format='json', expensive=None):
+def register(key, version, description=None, format='json', expensive=None, compare_key=None):
```

Comment:
@jbradberry I'm extending this a bit so that, when needed, we can customize how we compare entries. The _key_ function for job events is [here](https://github.com/ansible/awx/pull/9856/files#diff-5e7230fef85166a18967e2216535cd734d66de8b26983c303981e3c82b6fa1afR383).